src/alloc/AllocBenchmarks.jl

```diff
@@ -0,0 +1,58 @@
+module AllocBenchmarks
+
+using BenchmarkTools
+
+const SUITE = BenchmarkGroup()
+
+function perf_alloc_many_arrays()
+    for _ in 1:10000000
+        # global to ensure that this is heap allocated
+        global a = []
```

Comment:
Talking with @vtjnash, we don't think the inferencebarrier will be enough.

We also think that the overhead of the atomic fence here might be too high..

maybe switch to allocating a thousand things per alloc, like something like this:
```suggestion
        global a = [[] for _ in 1:1000]
```

Reply:
done